pwndbg/dbg/gdb.py

```diff
@@ -14,9 +14,24 @@
 
 import pwndbg
 import pwndbg.gdblib
+import pwndbg.gdblib.events
 from pwndbg.gdblib import gdb_version
 from pwndbg.gdblib import load_gdblib
 
+T = TypeVar("T")
+
+
+# We pass the responsibility of event handling to gdblib.
+GDBLIB_EVENT_MAPPING = {
+    pwndbg.dbg_mod.EventType.EXIT: pwndbg.gdblib.events.exit,
+    pwndbg.dbg_mod.EventType.CONTINUE: pwndbg.gdblib.events.cont,
+    pwndbg.dbg_mod.EventType.START: pwndbg.gdblib.events.start,
+    pwndbg.dbg_mod.EventType.STOP: pwndbg.gdblib.events.stop,
+    pwndbg.dbg_mod.EventType.NEW_MODULE: pwndbg.gdblib.events.new_objfile,
+    pwndbg.dbg_mod.EventType.MEMORY_CHANGED: pwndbg.gdblib.events.mem_changed,
+    pwndbg.dbg_mod.EventType.REGISTER_CHANGED: pwndbg.gdblib.events.reg_changed,
+}
```

Comment:
> In theory we could have just pwndbg.dbg.events.X defined based on the debugger?

I feel that, yes, we could, in the same way we could do `pwndbg.dbg.IS_GDB` to optimize `pwndbg.dbg.is_gdblib_available()`. Though considering there's only about ~50 uses of this function, all during initialization, I feel optimizing this in terms of time might not be all that crucial right now? We could leave that for later, since the interfaces can be renamed with a simple `sed` later on anyway